### PR TITLE
[Paracel-15]. Running outside.

### PR DESCRIPTION
**Pull request by xunzhang**
Usage:
- Compiling
`cd PARACEL_HOME;`
`mkdir build; cd build;`
`cmake -DCMAKE_INSTALL_PREFIX=/opt/paracel; make -j$(nproc); make install`

- Running
`cd anywhere;`
`/opt/paracel/prun.py -w 1 -p 1 -c demo.json /opt/paracel/bin/wc`

- Getting data.txt
`python /opt/paracel/tool/datagen.py -m wc -o data.txt`

  `demo.json`
  `{`
   ` "input" : "data.txt",`
    `"output" : "./wc_output/",`
    `"topk" : 20, `
    `"handle_file" : "/opt/paracel/lib/libwc_update.dylib",`
    `"update_function" : "wc_updater",`
    `"filter_function" : "wc_filter"`
  `}`

### CODE REVIEW

**Comment by xunzhang** [comment]
cc @zzl0 @musicbug

**Comment by xunzhang** [comment]
Document updated at: http://paracel.io/docs/api_reference.html#driver & http://paracel.io/docs/quick_tutorial.html

**Comment by xunzhang** [comment]
also cc @windreamer plz review, thanks

File: prun.py [hunk 1]
```diff
@@ -30,9 +30,19 @@
 logging.basicConfig(filename='paracelrun_log', format = '%(asctime)s : %(levelname)s : %(message)s', level = logging.INFO)
 logger = logging.getLogger(__name__)
 
-PARACEL_INSTALL_PREFIX='./local/'
+PARACEL_INSTALL_PREFIX = os.path.dirname(os.path.abspath(__file__))
 #PARACEL_INSTALL_PREFIX='/nfs/xxx/paracel/local'
-#PARACEL_INSTALL_PREFIX='/usr/local'
+#PARACEL_INSTALL_PREFIX='./local'
```
Comment (windreamer) [inline prun.py]: I don't this part is necessary. Once successfully installed, `$(PARACEL_INSTALL_PREFIX)/lib` should has been added to `/etc/ld.so.conf`

File: mrun [hunk 2]
```diff
@@ -1,3 +1,2 @@
 #! /bin/sh
-
-exec python "mesos_scheduler.py" --mpi "$@"
+exec python "$PARACEL_INSTALL_PREFIX/mesos_scheduler.py" --mpi "$@"
```
Comment (windreamer) [inline mrun]: try `$(readlink -f $(dirname "$0"))`

File: prun.py (hunk 1 above)
Comment (xunzhang) [inline prun.py]: Thanks for reply. What about the case users install paracel not in system folder? `LD_LIBRARY_PATH` is one way to update system linking configuration. What's your suggested way?

Comment (windreamer) [inline prun.py]: use `/etc/ld.so.conf` or `/etc/ld.so.conf.d`

File: mrun (hunk 2 above)
Comment (xunzhang) [inline mrun]: Yes, this way came to my mind firstly. But the concern here is usage of `readlink` is very different from osx to linux.

Comment (windreamer) [inline mrun]: but `PARACEL_INSTALL_PREFIX` only available during `make`.
I think just `$(dirname "$0")` is enough

File: prun.py (hunk 1 above)
Comment (xunzhang) [inline prun.py]: ok, I will try to modify soon.

File: mrun (hunk 2 above)
Comment (xunzhang) [inline mrun]: `PARACEL_INSTALL_PREFIX` is set to its right value inside `prun.py`. `mrun` can be only called inside `prun.py`, I mean they are always in the same process with same env. So don't worry about the logic. Anyway, I will try to use `dirname`.

**Comment by xunzhang** [comment]
@windreamer , I update the commit according our discussion, plz review again.

**Pull request merged**